chartfx-dataset/src/main/java/de/gsi/dataset/spi/DataSetBuilder.java

```diff
@@ -1,32 +1,65 @@
 package de.gsi.dataset.spi;
 
+import java.lang.reflect.InvocationTargetException;
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import java.util.Map.Entry;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 import de.gsi.dataset.AxisDescription;
 import de.gsi.dataset.DataSet;
+import de.gsi.dataset.DataSetMetaData;
+import de.gsi.dataset.EditableDataSet;
 import de.gsi.dataset.utils.AssertUtils;
 
+/**
+ * Helps allocating new DataSets in a flexible and efficient way.
+ * Values, Errors and Metadata can be added along with requirements for the resulting DataSet.
+ * Values are automatically converted between float and double if needed.
+ * <p>
+ * Currently the builder generates the following DataSet types:
+ * <ul>
+ * <li>dim:2, useFloat=false, errors=false: {@link DefaultDataSet}
+ * <li>dim:2, useFloat=false, errors=true: {@link DefaultErrorDataSet}
+ * <li>dim:2, useFloat=true, errors=false: {@link FloatDataSet}
+ * <li>dim:3+, useFloat=false, errors=false: {@link MultiDimDoubleDataSet}
+ * <li>All other combinations throw a {@code UnsupportedOperationException}
+ * </ul>
+ * 
+ * @author Alexander Krimm
+ */
 public class DataSetBuilder {
-    private static final int MAX_AXIS_DIMENSION = 1000;
+    static final Logger LOGGER = LoggerFactory.getLogger(DataSetBuilder.class);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use explicit scoping instead of the default package private level](https://app.codacy.com/manual/GSI/chart-fx/pullRequest?prid=5481451)